vector/src/main/java/im/vector/activity/VectorRoomActivity.java

```diff
@@ -2976,24 +2996,17 @@ public void onClick(View view) {
         }
     }
 
-
     /**
      * Tell if the user can send a message in this room.
      *
      * @return true if the user is allowed to send messages in this room.
      */
-    private boolean canSendMessages() {
-        boolean canSendMessage = false;
-
-        if ((null != mRoom) && (null != mRoom.getState())) {
-            canSendMessage = true;
-            PowerLevels powerLevels = mRoom.getState().getPowerLevels();
-
-            if (null != powerLevels) {
-                canSendMessage = powerLevels.maySendMessage(mMyUserId);
-            }
+    private boolean canSendMessages(@NonNull final RoomState state) {
+        boolean canSendMessage = !state.isVersioned();
+        if (canSendMessage) {
+            final PowerLevels powerLevels = state.getPowerLevels();
+            canSendMessage = (powerLevels != null && powerLevels.maySendMessage(mMyUserId));
```

Comment:
With the previous algo, if powerLevels is null, user can send message. And now he cannot. Maybe it is better now, maybe it is not...

Reply:
I think it's better :-p